cryptoprotocol/src/main/java/org/cryptimeleon/incentivesystem/cryptoprotocol/model/proofs/CommitmentWellformednessProtocol.java

```diff
@@ -0,0 +1,139 @@
+package org.cryptimeleon.incentivesystem.cryptoprotocol.model.proofs;
+
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.cryptimeleon.craco.protocols.CommonInput;
+import org.cryptimeleon.craco.protocols.SecretInput;
+import org.cryptimeleon.craco.protocols.arguments.sigma.schnorr.DelegateProtocol;
+import org.cryptimeleon.craco.protocols.arguments.sigma.schnorr.LinearStatementFragment;
+import org.cryptimeleon.craco.protocols.arguments.sigma.schnorr.SendThenDelegateFragment;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.IncentivePublicParameters;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderPublicKey;
+import org.cryptimeleon.math.structures.rings.zn.Zn;
+
+import java.math.BigInteger;
+
+/**
+ * A protocol for proof of the well-formedness of the preliminary commitment in the initial user token as defined in the Issue<->Join protocol of
+ * the Cryptimeleon incentive system.
+ *
+ * In particular, it proves knowledge of (usk, t, z, 1/u, eskUsr, dsrnd0, dsrnd1) in Z_p^7 s.t.
+ * I    upk = w^usk
+ * II   ( C^pre_0 ) ^ 1/u = h1^usk * h2^esk_usr * h3^dsrnd0 * h4^dsrnd1 * h6^z * h7^t
+ * III  g1 = ( C^pre_1 ) ^ 1/u
+ *
+ * Note that instead of u, 1/u is used to avoid implementation of double exponent proofs.
+ *
+ * The common input for the interactive protocol for that proof is split into two parts:
+ * the static part consists of the variables that are the same for every interaction, namely
+ * w, h1, h2, h3, h4, h6, h7, g1 .
+ * These values are stored in the CommitmentWellformednessProof object.
+ * On the other hand, there is the non-static part of the common input which comprises
+ * upk, c0Pre, c1Pre
+ */
+@Value
```

Comment:
```suggestion
@Value
```
@Value is not suited for this class

Reply:
What is the problem exactly?

Reply:
I think the suggestion is outdated